disabled-tests-inspector/src/main/java/io/quarkus/ts/DisabledTestAnalyserService.java

```diff
@@ -0,0 +1,217 @@
+
+                    if (reasonMatcher.find()) {
+                        reason = reasonMatcher.group(1);
+                    }
+
+                    if (reason == null) {
+                        Matcher explicitReasonmatcher = EXPLICIT_REASON_PATTERN.matcher(annotationContent);
+                        if (explicitReasonmatcher.find()) {
+                            reason = explicitReasonmatcher.group(1);
+                        }
+                    }
+                    issueLink = extractIssueLink(annotationContent);
+                }
+
+                if (issueLink == null && previousLine.startsWith("//")) {
+                    issueLink = extractIssueLink(previousLine);
+                }
+
+                if (issueLink == null && currentLine.startsWith("//")) {
+                    issueLink = extractIssueLink(currentLine);
+                }
+
+                if (reason == null) {
+                    reason = getDisablingReasonComment(previousLine);
+                }
+
+                if (reason == null) {
+                    reason = getDisablingReasonComment(currentLine);
+                }
+
+                annotationTypes.add(annotationType);
+                reasons.add(reason);
+                issueLinks.add(issueLink);
+            }
+
+            if (insideDisabledBlock && (testMethodMatcher.lookingAt() || (classMatcher.lookingAt()))) {
+                for (int i = 0; i < annotationTypes.size(); i++) {
+                    disabledTests.add(new DisabledTest(
+                            currentTestMethod != null ? currentTestMethod : "All methods",
+                            currentClass,
+                            annotationTypes.get(i),
+                            reasons.get(i),
+                            issueLinks.get(i),
+                            fileUrl,
+                            isGitHubIssueClosed(issueLinks.get(i))
+                    ));
+
+                    String moduleName = extractModuleName(ghContent.getPath());
+                    moduleStats.putIfAbsent(moduleName, new DisabledTestsModuleStats());
+                    moduleStats.get(moduleName).incrementAnnotation(annotationTypes.get(i));
+                }
+
+                annotationTypes.clear();
+                reasons.clear();
+                issueLinks.clear();
+                insideDisabledBlock = false;
+            }
+            previousLine = currentLine;
+        }
+        return disabledTests;
+    }
+
+    private String extractIssueLink(String text) {
+        Matcher issueMatcher = ISSUE_PATTERN.matcher(text);
+        return issueMatcher.find() ? issueMatcher.group(1) : null;
+    }
+
+    private String getDisablingReasonComment(String line) {
+        if (line.contains("//")) {
+            return line.substring(line.indexOf("//") + 2).trim();
+        }
+        return null;
+    }
+
+    private boolean isGitHubIssueClosed(String issueLink) {
```

Comment:
remote calls are expensive and we had issues in the past that forced us to disable many tests, you should cache result of this call instead of making remote call again and again for the same issue